### PR TITLE
telemetry, session: add telemetry collector for placement policies

### LINKED ISSUE #33188: support telemetry for "Placement Rules in SQL"
## Enhancement

We have introduced "Placement Rules in SQL" in #18030, it is necessary to add the usage of this feature for telemetry so we have a better statistics for the improvements in the future.

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve?

Issue Number: close #33188 

Problem Summary:

### What is changed and how it works?

Add telemetry support.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: telemetry/data_feature_usage.go [hunk 1]
```diff
@@ -33,32 +33,75 @@ type featureUsage struct {
...
+	CachedTable          bool                  `json:"cachedTable"`
+	AutoCapture          bool                  `json:"autoCapture"`
+	PlacementPolicyUsage *placementPolicyUsage `json:"placementPolicy"`
+}
+
+type placementPolicyUsage struct {
+	NumPlacementPolicies      uint64
+	NumDBWithPolicies         uint64
+	NumTableWithPolicies      uint64
+	NumPartitionWithPoliciess uint64
```
Comment (lcwangchao) [inline telemetry/data_feature_usage.go:48]:
```suggestion
	NumPartitionWithPolicies uint64
```

File: telemetry/data_feature_usage.go [hunk 2]
```diff
@@ -33,32 +33,75 @@ type featureUsage struct {
...
+			if tbInfo.Meta().PlacementPolicyRef != nil {
+				usage.PlacementPolicyUsage.NumTableWithPolicies++
+			}
+			partitions := tbInfo.Meta().GetPartitionInfo()
+			if partitions == nil {
+				continue
+			}
+			for _, partitionInfo := range partitions.Definitions {
+				if partitionInfo.PlacementPolicyRef != nil {
+					usage.PlacementPolicyUsage.NumPartitionWithPoliciess++
```
Comment (lcwangchao) [inline telemetry/data_feature_usage.go:98]: What is `NumPartitionWithPolicies` mean ? If a table has placement policy, all its partitions should have placement policies.

File: telemetry/data_feature_usage.go (hunk 1 above)
Comment (bb7133) [reply telemetry/data_feature_usage.go:48]: Resolved, thanks.

File: telemetry/data_feature_usage.go (hunk 2 above)
Comment (bb7133) [reply telemetry/data_feature_usage.go:98]: I've updated it to `NumPartitionWithExplicitPolicies` with a comment.

**Review by lcwangchao — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**